tests/lib/__init__.py

```diff
@@ -469,6 +469,28 @@ def main():
     return version_pkg_path
 
 
+def _create_test_package_with_srcdir(script, name='version_pkg', vcs='git'):
+    script.scratch_path.join(name).mkdir()
+    version_pkg_path = script.scratch_path / name
+    subdir_path = version_pkg_path.join('subdir')
+    subdir_path.mkdir()
+    src_path = subdir_path.join('src')
+    src_path.mkdir()
+    pkg_path = src_path.join('pkg')
+    pkg_path.mkdir()
```

Comment:
might be cleaner to directly use `os.makedirs`, cf https://github.com/pypa/pip/blob/develop/tests/unit/test_req.py#L43-L44